### PR TITLE
Add more routing unit tests

### LINKED ISSUE #832: Refactor large routing functions into smaller ones and add unit tests
I came to the conclusion together with @JoostBuitink that it would be helpful for understanding and bug-fixing the code that certain large functions are split up into smaller ones which are then unit tested. I have Identified:

- `kinwave_river_update!`
- `local_inertial_river_update!`
- `local_inertial_update_fluxes!`
- `local_inertial_update_water_depth!`

I will take into account the conclusions from https://github.com/Deltares/Wflow.jl/pull/793.

### PR DESCRIPTION
**Pull request by SouthEndMusic**
## Issue addressed
Fixes https://github.com/Deltares/Wflow.jl/issues/832

## Explanation
Explain how you addressed the bug/feature request, what choices you made and why.

## Checklist
- [x] Updated tests or added new tests
- [x] Branch is up to date with `master`
- [x] Tests & pre-commit hooks pass
- [x] Updated documentation if needed
- [x] Updated changelog.qmd if needed

## Additional Notes (optional)

Some notes to the reviewer: I let Copilot do most of the refactoring work. Points of attention:
- Whether the new functions have sensible names and are in the right file
- Whether the values in the tests make sense. I did make sure to reach the most complex code paths

### CODE REVIEW

File: Wflow/src/routing/surface_kinwave.jl [hunk 1]
```diff
@@ -367,21 +367,71 @@ function update!(model::KinWaveOverlandFlow, domain::DomainLand, dt::Float64)
     return nothing
 end
 
+"""
+run reservoir model and copy reservoir outflow to inflow (qin) of
+downstream river cell
+"""
```
Comment (vers-w) [inline Wflow/src/routing/surface_kinwave.jl]: Docstring fits on one line:

```suggestion
"Run reservoir model and copy reservoir outflow to inflow (qin) of downstream river cell"
```

File: Wflow/src/routing/surface_kinwave.jl [hunk 2]
```diff
@@ -367,21 +367,71 @@ function update!(model::KinWaveOverlandFlow, domain::DomainLand, dt::Float64)
...
+function update_reservoir!(
+    reservoir::Reservoir,
```
Comment (vers-w) [inline Wflow/src/routing/surface_kinwave.jl]: Consistent with suggestions in PR 833.

```suggestion
function update_reservoir_model!(
    reservoir_model::Reservoir,
```

File: Wflow/src/routing/surface_local_inertial.jl [hunk 3]
```diff
@@ -309,188 +306,282 @@ function local_inertial_river_update!(
...
+    model::LocalInertialRiverFlow{R, F},
+    domain::DomainRiver,
+    dt::Float64,
+) where {R, F <: Nothing} = nothing
+
+"""
+Update reservoir boundary conditions for the local inertial river flow model.
+"""
+function update_boundary_conditions_reservoir!(
+    model::LocalInertialRiverFlow,
```
Comment (vers-w) [inline Wflow/src/routing/surface_local_inertial.jl]: Suggest to use a similar approach here as for the reservoir in the kinematic wave river flow model. 

```suggestion
function update_reservoir_model!(
    reservoir_model::Reservoir,
    river_flow_model::LocalInertialRiverFlow,
```

File: Wflow/src/routing/surface_local_inertial.jl [hunk 4]
```diff
@@ -309,188 +306,282 @@ function local_inertial_river_update!(
...
+    dt::Float64,
+    dt_forcing::Float64,
+)
+    (; edges_at_node) = domain.river.network
     (; reservoir) = model.boundary_conditions
     inds_reservoir = domain.reservoir.network.river_indices
-    if !isnothing(reservoir)
-        res_bc = reservoir.boundary_conditions
-    end
+    isnothing(reservoir) && return nothing
```
Comment (vers-w) [inline Wflow/src/routing/surface_local_inertial.jl]: Could be part of dispatch, see comment above.

File: Wflow/src/routing/surface/surface_local_inertial.jl [hunk 5]
```diff
@@ -820,6 +911,82 @@ function update!(
     return nothing
 end
 
+"""
+Update flow for a single direction in the local inertial overland flow model.
+`is_x_direction`: true for x-direction (xu/xd), false for y-direction (yu/yd)
+"""
+@inline function update_directional_flow!(
```
Comment (vers-w) [inline Wflow/src/routing/surface/surface_local_inertial.jl:943]: Just checking: this change (using `@inline`) did not affect multi-threading performance?

File: Wflow/src/routing/surface_local_inertial.jl [hunk 6]
```diff
@@ -949,96 +1038,200 @@ function update_inflow_reservoir!(
...
+    yd = indices.yd[i]
+    xd = indices.xd[i]
+
+    return (
+        sum_at(river.variables.q, edges_at_node.src[river_idx]) -
+        sum_at(river.variables.q, edges_at_node.dst[river_idx]) + land.variables.qx[xd] -
+        land.variables.qx[i] + land.variables.qy[yd] - land.variables.qy[i] +
+        land.boundary_conditions.runoff[i] -
+        river.boundary_conditions.abstraction[river_idx]
+    ) * dt
```
Comment (vers-w) [inline Wflow/src/routing/surface_local_inertial.jl]: This is a bit large. 
Why not move this up, maybe split it a bit, assign result to and return a variable?

File: Wflow/test/reservoir.jl [hunk 7]
```diff
@@ -119,6 +119,57 @@ end
...
+    using Graphs: DiGraph, add_edge!
+
+    n = 1
+    reservoir = Wflow.Reservoir(;
+        boundary_conditions = Wflow.ReservoirBC(;
+            n,
+            external_inflow = [-1.0],
+            inflow_overland = [0.02],
+            inflow_subsurface = [0.04],
+            inflow = [0.06],
```
Comment (vers-w) [inline Wflow/test/reservoir.jl]: Suggest to use a quite higher value for `inflow`, for example > 10.0.

File: Wflow/test/reservoir.jl [hunk 8]
```diff
@@ -119,6 +119,57 @@ end
...
+            evaporation = [0.1],
+        ),
+        parameters = Wflow.ReservoirParameters(;
+            id = [1],
+            storfunc = [Wflow.ReservoirProfileType.linear],
+            outflowfunc = [Wflow.ReservoirOutflowType.simple],
+            area = [6.0e4],
+            threshold = [0.0],
+            b = [0.0],
+            e = [0.0],
```
Comment (vers-w) [inline Wflow/test/reservoir.jl]: You need to set parameters here that belong to the simple reservoir concept:
`maxrelease`
`demand`
`targetminfrac`
`targetfullfrac`
`maxstorage`

File: Wflow/test/reservoir.jl [hunk 9]
```diff
@@ -119,6 +119,57 @@ end
...
+        ),
+        variables = Wflow.ReservoirVariables(;
+            waterlevel = [1.0],
+            storage = [4.5e7],
+            outflow = [3.0],
+            outflow_av = [2.0],
+            outflow_obs = [1.0],
```
Comment (vers-w) [inline Wflow/test/reservoir.jl]: Suggest to use a missing value here, then `outflow_obs` will not be used, and the simple reservoir functionality is tested.

```suggestion
            outflow_obs = [Wflow.MISSING_VALUE],
```

File: Wflow/test/reservoir.jl [hunk 10]
```diff
@@ -119,6 +119,57 @@ end
...
+            storfunc = [Wflow.ReservoirProfileType.linear],
+            outflowfunc = [Wflow.ReservoirOutflowType.simple],
+            area = [6.0e4],
+            threshold = [0.0],
+            b = [0.0],
+            e = [0.0],
+        ),
+        variables = Wflow.ReservoirVariables(;
+            waterlevel = [1.0],
+            storage = [4.5e7],
```
Comment (vers-w) [inline Wflow/test/reservoir.jl]: These values are not in line with the `area` of the reservoir.

File: Wflow/test/reservoir.jl [hunk 11]
```diff
@@ -119,6 +119,57 @@ end
...
+            outflow = [3.0],
+            outflow_av = [2.0],
```
Comment (vers-w) [inline Wflow/test/reservoir.jl]: These variables and `actevap` do not have to be set, these are computed (output variables).

File: Wflow/test/routing_process.jl [hunk 12]
```diff
@@ -207,6 +207,357 @@ end
...
+        boundary_conditions = Wflow.RiverFlowBC(;
+            n,
+            external_inflow = [-0.1],
+            reservoir = Wflow.Reservoir(;
+                boundary_conditions = Wflow.ReservoirBC(; n, external_inflow = [0.02]),
+                parameters = Wflow.ReservoirParameters(;
+                    id = [1],
+                    storfunc = [Wflow.ReservoirProfileType.linear],
+                    outflowfunc = [Wflow.ReservoirOutflowType.simple],
+                    area = [2500.0],
```
Comment (vers-w) [inline Wflow/test/routing_process.jl]: Add missing parameters for simple reservoir concept. Now there is no output for the reservoir.

File: Wflow/test/routing_process.jl [hunk 13]
```diff
@@ -207,6 +207,357 @@ end
...
+        ),
+        parameters = Wflow.RiverFlowParameters(;
+            flow = Wflow.ManningFlowParameters(;
+                n,
+                beta = 0.6,
+                slope = [0.01],
+                mannings_n = [0.03],
+                alpha_pow = 0.4,
+                alpha_term = [0.5],
+                alpha = [5.0],
```
Comment (vers-w) [inline Wflow/test/routing_process.jl]: Are these two values based on computations in the code (these are set in the `update!` function), and thus in agreement?

File: Wflow/test/routing_process.jl [hunk 14]
```diff
@@ -207,6 +207,357 @@ end
...
+            ),
+            bankfull_depth = [10.0],
+        ),
+        variables = Wflow.FlowVariables(; n, q = [0.2]),
```
Comment (vers-w) [inline Wflow/test/routing_process.jl]: Quite low value, would make use of a higher value for a 30 m wide river with a slope of 0.01, for example ~30.0 m3/s. Also, a `q` value seems missing for the other river/reservoir cell?

File: Wflow/test/routing_process.jl [hunk 15]
```diff
@@ -207,6 +207,357 @@ end
     @test new_material == [8.0, 0.0, 0.0, 10.0, 0.0, 40.0]
 end
 
+@testitem "unit: kinwave_river_update!" begin
+    using Graphs: DiGraph, add_edge!
```
Comment (vers-w) [inline Wflow/test/routing_process.jl:269]: Would be nice to have a short description of this test, for example where the reservoir is located on the graph.

File: Wflow/test/routing_process.jl [hunk 16]
```diff
@@ -207,6 +207,357 @@ end
...
+            reservoir_indices = [1],
+        ),
+        parameters = Wflow.RiverParameters(; flow_width = [30.0], flow_length = [800.0]),
+    )
+    dt = 1200.0
+    dt_forcing = 86400.0
+
+    Wflow.kinwave_river_update!(model, domain, dt, dt_forcing)
+
+    @test model.variables.q ≈ [0.1598124775930105]
```
Comment (vers-w) [inline Wflow/test/routing_process.jl]: What about adding more variables to test, for example related to the reservoir model?

File: Wflow/test/routing_process.jl [hunk 17]
```diff
@@ -207,6 +207,357 @@ end
...
+            graph,
+            order_of_subdomains = [[1]],
+            order_subdomain = [[1]],
+            subdomain_indices = [[1]],
+            upstream_nodes = [[]],
+            reservoir_indices = [1],
+        ),
+        parameters = Wflow.RiverParameters(; flow_width = [30.0], flow_length = [800.0]),
+    )
+    dt = 1200.0
```
Comment (vers-w) [inline Wflow/test/routing_process.jl]: Is this timestep in agreement with the stable timestep (courant number)? Maybe better to just compute it for this test case.

File: Wflow/test/routing_process.jl [hunk 18]
```diff
@@ -207,6 +207,357 @@ end
...
+        boundary_conditions = Wflow.RiverFlowBC(;
+            n,
+            external_inflow = [-1.0, -1.0],
+            inwater = [100.0, 100.0],
+            reservoir = Wflow.Reservoir(;
+                boundary_conditions = Wflow.ReservoirBC(;
+                    n,
+                    external_inflow = [-1.0],
+                    inflow_overland = [3000.0],
+                    inflow_subsurface = [5000.0],
```
Comment (vers-w) [inline Wflow/test/routing_process.jl]: These values are very high.

```suggestion
                    inflow_overland = [0.10],
                    inflow_subsurface = [0.25],
```

File: Wflow/test/routing_process.jl [hunk 19]
```diff
@@ -207,6 +207,357 @@ end
...
+end
+
+@testitem "unit: local_inertial_river_update!" begin
+    n = 2
+    model = Wflow.LocalInertialRiverFlow(;
+        timestepping = Wflow.TimeStepping(),
+        boundary_conditions = Wflow.RiverFlowBC(;
+            n,
+            external_inflow = [-1.0, -1.0],
+            inwater = [100.0, 100.0],
```
Comment (vers-w) [inline Wflow/test/routing_process.jl]: Quite high values.

```suggestion
            inwater = [1.0, 1.0],
```

File: Wflow/test/routing_process.jl [hunk 20]
```diff
@@ -207,6 +207,357 @@ end
...
+                    id = [1, 2],
+                    storfunc = [Wflow.ReservoirProfileType.linear],
+                    outflowfunc = [Wflow.ReservoirOutflowType.simple],
+                    area = [5.0e6, 4.0e6],
+                    maxrelease = [10.0, 10.0],
+                    demand = [1.5, 1.5],
+                    targetminfrac = [0.3, 0.3],
+                    targetfullfrac = [0.1, 0.1],
+                    maxstorage = [Inf, Inf],
+                    threshold = [0.0, 0.0],
```
Comment (vers-w) [inline Wflow/test/routing_process.jl]: Why `Inf` for `maxstorage`? `threshold` is not required for simple reservoir.

File: Wflow/test/routing_process.jl [hunk 21]
```diff
@@ -207,6 +207,357 @@ end
...
+                ),
+                variables = Wflow.ReservoirVariables(;
+                    waterlevel = [1.0, 1.0],
+                    storage = [2.5e5, 2.5e5],
```
Comment (vers-w) [inline Wflow/test/routing_process.jl]: Only these two variables are required, the others are computed during the update.

Storage is not consistent with  `waterlevel` and `area`.

File: Wflow/test/routing_process.jl [hunk 22]
```diff
@@ -207,6 +207,357 @@ end
...
+                ),
+                parameters = Wflow.ReservoirParameters(;
+                    id = [1, 2],
+                    storfunc = [Wflow.ReservoirProfileType.linear],
+                    outflowfunc = [Wflow.ReservoirOutflowType.simple],
+                    area = [5.0e6, 4.0e6],
+                    maxrelease = [10.0, 10.0],
+                    demand = [1.5, 1.5],
+                    targetminfrac = [0.3, 0.3],
+                    targetfullfrac = [0.1, 0.1],
```
Comment (vers-w) [inline Wflow/test/routing_process.jl]: Would reverse the values here.

File: Wflow/test/routing_process.jl [hunk 23]
```diff
@@ -207,6 +207,357 @@ end
...
+    Wflow.kinwave_river_update!(model, domain, dt, dt_forcing)
+
+    @test model.variables.q ≈ [0.1598124775930105]
+    @test model.variables.h[1] ≈ 0.055464507410878765
+    @test model.variables.storage[1] ≈ 1331.1481778610903
+    @test model.variables.q_av[1] ≈ 191.7749731116126
+end
+
+@testitem "unit: local_inertial_river_update!" begin
+    n = 2
```
Comment (vers-w) [inline Wflow/test/routing_process.jl]: Would be nice to have a short description of the test, nodes/edges, location reservoir etc. Could also be in a simple diagram as done in groundwater.jl.

File: Wflow/test/routing_process.jl [hunk 24]
```diff
@@ -207,6 +207,357 @@ end
...
+            parameters = Wflow.RiverParameters(;
+                flow_width = [10.0, 10.0],
+                flow_length = [100.0, 100.0],
+            ),
+        ),
+        reservoir = Wflow.DomainReservoir(;
+            network = Wflow.NetworkReservoir(; river_indices = [1]),
+        ),
+    )
+    dt = 1000.0
```
Comment (vers-w) [inline Wflow/test/routing_process.jl]: Consistent with stable timestep? Maybe better to just compute it.

File: Wflow/test/routing_process.jl [hunk 25]
```diff
@@ -207,6 +207,357 @@ end
...
+                boundary_conditions = Wflow.ReservoirBC(;
+                    n,
+                    external_inflow = [-1.0],
+                    inflow_overland = [3000.0],
+                    inflow_subsurface = [5000.0],
+                    precipitation = [2.0],
+                    evaporation = [1.0],
+                    inflow = [0.0],
+                ),
+                parameters = Wflow.ReservoirParameters(;
```
Comment (vers-w) [inline Wflow/test/routing_process.jl:376]: Why defining two reservoirs with only two river nodes? For each reservoir node (and edge) local inertial flow computations are not active and replaced by the reservoir model.

File: Wflow/test/routing_process.jl [hunk 26]
```diff
@@ -207,6 +207,357 @@ end
...
+            q = [0.0, 1e-4],
+        ),
+        floodplain = Wflow.FloodPlain(;
+            parameters = Wflow.FloodPlainParameters(;
+                profile = Wflow.FloodPlainProfile(;
+                    depth = [10.0, 10.0],
+                    storage = [1e5 1e6; 1e5 1e6],
+                    width = [100.0 100.0; 100.0 100.0],
+                    a = [1e3 1e3; 1e3 1e3],
+                    p = [400.0 400.0; 400.0 400.0],
```
Comment (vers-w) [inline Wflow/test/routing_process.jl]: The `profile` parameter values don't seem correct. `depth` should start at 0.0 and increase, mostly steps of 0.5 m are used. `storage`, `a` and `p` are cumulative, and `width` should start at river `width` and increase (otherwise there is no floodplain). The different parameters (values) should also be in agreement.

File: Wflow/test/routing_process.jl [hunk 27]
```diff
@@ -207,6 +207,357 @@ end
...
+
+    @test model.boundary_conditions.reservoir.variables.storage[1] ≈ 7.391913765967477e6
+    @test model.boundary_conditions.reservoir.variables.waterlevel[1] ≈ 2.428382753193495
+    @test model.boundary_conditions.reservoir.variables.outflow[1] ≈ 0.00018509186397934759
+    @test model.boundary_conditions.reservoir.boundary_conditions.inflow[1] ≈
+          7.141856080688971e6
+    @test model.boundary_conditions.reservoir.variables.outflow_av[1] ≈ 0.1850918639793476
+    @test model.boundary_conditions.reservoir.variables.actevap[1] ≈ 0.011574074074074073
+    @test model.variables.q[1] ≈ 0.00018509186397934759
+    @test model.variables.q_av[1] ≈ -575303.5933591384
```
Comment (vers-w) [inline Wflow/test/routing_process.jl]: Suggest to make a more realistic test. This is a very large flow value that also flows in the upstream direction. The test could of course also be based on a wflow (test) model.

File: Wflow/test/routing_process.jl [hunk 28]
```diff
@@ -207,6 +207,357 @@ end
...
+
+    Wflow.update_water_depth_and_storage!(model, domain.river, dt)
+
+    @test model.variables.storage[2] ≈ 99000.0
+    @test model.variables.h[2] ≈ 99.0
+
+    Wflow.update_water_depth_and_storage!(model.floodplain, model, domain.river, dt)
+
+    @test model.variables.h ≈ [1.0, -79.7]
+    @test model.variables.storage ≈ [0.0, -79700.0]
```
Comment (vers-w) [inline Wflow/test/routing_process.jl]: Negative `h` and `storage` values, see also comment above about realistic test.

File: Wflow/test/routing_process.jl [hunk 29]
```diff
@@ -207,6 +207,357 @@ end
...
+                y_length = fill(900.0, n),
+            ),
+        ),
+    )
+    i = 1
+    dt = 60.0
+    is_x_direction = true
+
+    Wflow.update_directional_flow!(land, domain, i, dt, is_x_direction)
+    @test land.variables.qx_av[1] ≈ 26493.90166029366
```
Comment (vers-w) [inline Wflow/test/routing_process.jl]: Very large value, also for `qx`. This is overland flow, suggest to change the test, perhaps based on a wflow (test) model. It seems the parameter values are not in agreement.

File: Wflow/test/routing_process.jl [hunk 30]
```diff
@@ -207,6 +207,357 @@ end
...
+end
+
+@testitem "unit: local_inertial_update_water_depth!" begin
```
Comment (vers-w) [inline Wflow/test/routing_process.jl:717]: As for the other tests, suggest to check if parameter/variable values are in agreement, if `dt` value is in line with stable timestep etc. It may help to use a wflow (test) model to extract these values.

**Review by vers-w — CHANGES_REQUESTED** [review]
General comment for the added tests: would be good to have for some tests a bit more description about the test (for example network/graph and reservoir location etc), this could also be part of a simple diagram as done for example in groundwater.jl. Additionally, for the different tests the parameter/variable values should be in agreement, and have realistic input and output values, this could be also based on a wflow (test) model.

File: Wflow/src/routing/surface/surface_local_inertial.jl (hunk 5 above)
Comment (SouthEndMusic) [reply Wflow/src/routing/surface/surface_local_inertial.jl:943]: It did change the performance, but for the better. I don't know why, but I had to add this to resolve a large slow down.

File: Wflow/test/routing_process.jl (hunk 29 above)
Comment (SouthEndMusic) [reply Wflow/test/routing_process.jl]: I tried to fix it by using a part of the `sbm_river-land-local-inertial_config.toml`, but I only made it worse :')

Comment (SouthEndMusic) [reply Wflow/test/routing_process.jl]: I really need some more hands on help here

File: Wflow/test/reservoir.jl [hunk 31]
```diff
@@ -119,6 +119,55 @@ end
...
+            evaporation = [4.3],
+        ),
+        parameters = Wflow.ReservoirParameters(;
+            id = [1],
+            storfunc = [Wflow.ReservoirProfileType.linear],
+            outflowfunc = [Wflow.ReservoirOutflowType.simple],
+            area = [6.0e4],
+            threshold = [0.0],
+            b = [0.0],
+            e = [0.0],
```
Comment (vers-w) [inline Wflow/test/reservoir.jl]: nitpick: suggest to remove these parameters (not part of simple reservoir concept.

File: Wflow/test/reservoir.jl [hunk 32]
```diff
@@ -119,6 +119,55 @@ end
...
+            maxrelease = [63.0],
+            demand = [7.9],
+            targetminfrac = [0.003],
+            targetfullfrac = [1.0],
+            maxstorage = [7.16e7],
+        ),
+        variables = Wflow.ReservoirVariables(; waterlevel = zeros(n), storage = [7.1e7]),
```
Comment (vers-w) [inline Wflow/test/reservoir.jl]: These `waterlevel` and `storage` values are not aligned. For the simple reservoir concept `waterlevel` equals `storage` / `area`. Maybe easier to derive the values (parameters and variables) from a wflow (test) model?

File: Wflow/test/routing_process.jl [hunk 33]
```diff
@@ -260,6 +260,391 @@ end
...
+                    area = [1.79e6],
+                    maxrelease = [63.0],
+                    demand = [7.9],
+                    targetminfrac = [0.003],
+                    targetfullfrac = [1.0],
+                    maxstorage = [7.16e7],
+                ),
+                variables = Wflow.ReservoirVariables(;
+                    waterlevel = [40.0],
+                    storage = [7.15e7],
```
Comment (vers-w) [inline Wflow/test/routing_process.jl]: You only need to define `waterlevel` and `storage`, the other variables can be removed. The values are not aligned: storage = waterlevel * area.

File: Wflow/test/routing_process.jl [hunk 34]
```diff
@@ -260,6 +260,391 @@ end
...
+        ),
+        parameters = Wflow.RiverFlowParameters(;
+            flow = Wflow.ManningFlowParameters(;
+                n,
+                beta = 0.6,
+                slope = [0.01],
+                mannings_n = [0.03],
+                alpha_pow = 0.4,
+                alpha_term = [0.3008],
+                alpha = [1.628],
```
Comment (vers-w) [inline Wflow/test/routing_process.jl]: Are these values computed from the other parameters? Would be good to double check.

File: Wflow/test/routing_process.jl [hunk 35]
```diff
@@ -260,6 +260,391 @@ end
...
+            ),
+            bankfull_depth = [10.0],
+        ),
+        variables = Wflow.FlowVariables(; n, q = [58.3]),
```
Comment (vers-w) [inline Wflow/test/routing_process.jl]: Is this starting `q` value based on a wflow (test) model? If not, suggest to either use a wflow model as a starting point or for example base the starting value on the Mannings equation.

File: Wflow/test/routing_process.jl [hunk 36]
```diff
@@ -260,6 +260,391 @@ end
...
+            flow_length_at_edge = [1000.0, 1000.0],
+            flow_width_at_edge = [100.0, 100.0],
+        ),
+        variables = Wflow.LocalInertialRiverFlowVariables(;
+            n,
+            n_edges = 2,
+            q_av = [0.0, 0.0],
+            q_channel_av = [0.0, 0.0],
+            h = [1.0, 2.0],
+            q = [1e-4, 0.0],
```
Comment (vers-w) [inline Wflow/test/routing_process.jl]: Why is `q` 0.0 with an `h` of 2.0? 

It's also quite hard to figure out if other values are aligned. I would recommend to base this unit test on a wflow (test) model. For example, the `floodplain` has a `zb_max` of 1.0 while for the river this is 500.0, `mannings_n` has only one value for `floodplain` etc.

File: Wflow/test/routing_process.jl [hunk 37]
```diff
@@ -260,6 +260,391 @@ end
...
+            active_n = [1, 1],
+            active_e = [1, 1],
+            froude_limit = true,
+            h_thresh = 1e-3,
+            zb = [227.5, 313.3999938964844],
+            zb_max = [313.3999938964844],
+            bankfull_storage = [7979.0625, 32470.3125],
+            bankfull_depth = [1.0, 1.0],
+            mannings_n_sq = [0.001225, 0.0009],
+            mannings_n = [0.035, 0.03],
```
Comment (vers-w) [inline Wflow/test/routing_process.jl]: Only one value for `mannings_n_sq` should be defined (edge). Also it does not seem to be computed from the data/based on a wflow (test) model.

File: Wflow/test/routing_process.jl [hunk 38]
```diff
@@ -260,6 +260,391 @@ end
...
+            flow_width_at_edge = [30.0, 30.0],
+        ),
+        variables = Wflow.LocalInertialRiverFlowVariables(;
+            n,
+            n_edges = 2,
+            q = [0.00973319846856637, 0.1232553],
+            q_av = [0.0, 0.0],
+            q_channel_av = [0.0, 0.0],
+            h = [0.0083347, 0.007166168666],
+            storage = [66.50309237, 232.68773603860762],
```
Comment (vers-w) [inline Wflow/test/routing_process.jl]: Are `q`, `h` and `storage` in agreement? Again, would base this unit test on a wflow (test) model.

File: Wflow/test/routing_process.jl [hunk 39]
```diff
@@ -260,6 +260,391 @@ end
...
+        boundary_conditions = Wflow.RiverFlowBC(;
+            n,
+            external_inflow = [0.0, 0.0],
+            reservoir = nothing,
+        ),
+        parameters = Wflow.LocalInertialRiverFlowParameters(;
+            n,
+            ne = 2,
+            active_n = [1, 1],
+            active_e = [1, 1],
```
Comment (vers-w) [inline Wflow/test/routing_process.jl]: With two edges you would also expect two edge values (e.g. `zb_max`)?

**Review by vers-w — CHANGES_REQUESTED** [review]

File: Wflow/test/routing_process.jl (hunk 33 above)
Comment (SouthEndMusic) [reply Wflow/test/routing_process.jl]: I do need the other ones, otherwise they give propagating NaNs

**Review by vers-w — APPROVED** [review]
LGTM, as discussed I did update some of the routing unit tests.
